storage: avoid misleading consistency check message

tbg opened the pull request:
This was introduced in #21131 and caused each consistency check to
print an error message about a checksum mismatch.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21164)
<!-- Reviewable:end -->

RaduBerinde commented:
LGTM

tbg committed:
storage: avoid misleading consistency check message

This was introduced in #21131 and caused each consistency check to
print an error message about a checksum mismatch.

Release note: None
